### PR TITLE
:seedling: initial collect-assets script

### PR DESCRIPTION
**Pull request by savitharaghunathan**
Downloading kai analyzer & server

### CODE REVIEW

File: collect-assets.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  const response = await fetch(asset.browser_download_url);
+  if (!response.ok) {
+    throw new Error(`Failed to download ${asset.name}: ${response.statusText}`);
+  }
+
+  const fileStream = createWriteStream(assetPath);
+  await new Promise((resolve, reject) => {
+    response.body.pipe(fileStream);
+    response.body.on("error", reject);
+    fileStream.on("finish", resolve);
```
Comment (sjd78) [inline collect-assets.js:65]: Using node's `fetch()`...
```suggestion
    const reader = Readable.fromWeb(response.body);
    reader.pipe(fileStream);
    reader.on("error", reject);
    fileStream.on("finish", resolve);
```

**Review by sjd78 — COMMENTED** [review]
Looks like a great base to work from!

For the location of the script (and any future scripts), I'm thinking a `scripts/` would be appropriate.  Doesn't really matter right now since we can move things around when script #2 is created.

File: collect-assets.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
+import {
```
[the remark this answers is not in the document]
Comment (savitharaghunathan) [reply collect-assets.js:1]: @sjd78 thanks for the review. Can you help me with this comment? I am not sure how to add ```"type": "module" to the root package.json```

[the remark this answers is not in the document]
Comment (savitharaghunathan) [reply collect-assets.js:1]: nvm. I think I added it correctly to package.json

File: package.json [hunk 3]
```diff
@@ -82,8 +82,10 @@
   },
   "dependencies": {
     "js-yaml": "^4.1.0",
+    "node-fetch": "^3.3.2",
     "react": "^18.3.1",
     "react-dom": "^18.3.1",
+    "unzipper": "^0.12.3",
```
[the remark this answers is not in the document]
Comment (savitharaghunathan) [reply package.json]: resolved

**Comment by savitharaghunathan** [comment]
> For the location of the script (and any future scripts), I'm thinking a `scripts/` would be appropriate. Doesn't really matter right now since we can move things around when script 2 is created.

created a `scripts/` folder and moved the script into it

File: scripts/collect-assets.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    renameSync(oldPath, newPath);
+  });
+
+  console.log(`Extracted files for: ${asset.name}`);
+}
+
+(async function () {
+  try {
+    const releaseData = await getReleaseMetadata();
+    const commitId = releaseData.target_commitish;
```
Comment (sjd78) [inline scripts/collect-assets.js:87]: For some weird reason, the release endpoint doesn't return the sha for the release's commit.  Having the tag name is good, having the commit's sha is better, having both is best!

Here is a quick function to grab the tag's sha:
```js
async function getReleaseTagSha() {
  const url = `${GITHUB_API}/repos/${REPOSITORY}/commits/${RELEASE_TAG}`;
  const response = await fetch(url, {
    headers: {
      Accept: "application/vnd.github.sha",
    },
  });

  return await response.text();
}
```

Then just...
```suggestion
    const commitId = await getReleaseTagSha();
```

**Review by sjd78 — CHANGES_REQUESTED** [review]
2 more small changes and it'll be good to go.

File: scripts/collect-assets.js (hunk 4 above)
Comment (savitharaghunathan) [reply scripts/collect-assets.js]: resolved.thank you

**Review by sjd78 — APPROVED** [review]
LGTM

**Pull request merged**